tests/sqlparser_common.rs

```diff
@@ -7719,8 +7719,14 @@ fn parse_uncache_table() {
 #[test]
 fn parse_deeply_nested_parens_hits_recursion_limits() {
     let sql = "(".repeat(1000);
-    let res = parse_sql_statements(&sql);
-    assert_eq!(ParserError::RecursionLimitExceeded, res.unwrap_err());
+    let dialect = GenericDialect {};
+    let res = Parser::new(&dialect)
+        .try_with_sql(&sql)
+        .expect("tokenize to work")
+        .with_recursion_limit(38)
```

Comment:
The stack frame size increased in debug builds as this PR adds new local variables.
I fixed the test by decreasing recursion limit (39 is still raising an stack overflow error). @lustefaniak I could decrease it here as default value for logic https://github.com/getsynq/sqlparser-rs/blob/main/src/parser/mod.rs#L200, but not sure about it.

What is your advice?